src/Init/Data/Option/Instances.lean

```diff
@@ -144,6 +144,13 @@ none
   | none => b
   | some a => f a rfl
 
+/-- Partial filter. If `o : Option α`, `p : (a : α) → a ∈ x → Bool`, then `o.pfilter p` is
+the same as `o.filter p` but `p` is passed the proof that `a ∈ o`. -/
+@[inline] def pfilter (o : Option α) (p : (a : α) → a ∈ o → Bool) : Option α :=
```

Comment:
We're actually moving away from explicitly mentioning the `Membership` instance on `Option` anywhere in lemmas (it's not in `simp` normal form due to `Option.mem_def`), so please switch everything to use `o = some a` instead of `a ∈ o`.